4semester/26022016/26022016/Functions.fs

```diff
@@ -0,0 +1,44 @@
+﻿module Functions 
+
+/// Digitals multiplication
+let rec dMult n = 
+    if n < 10 then n else (n % 10) * dMult (n / 10)
+
+/// Index in list
+let indVarInList list var =
+    let rec innerRec ls v ind =
+        match ls with
+        | [] -> -1
+        | h :: t -> if v = h then ind else innerRec t v (ind + 1)
+    innerRec list var 0
+
+/// Reverses list (helper function)
+let reverseList list = 
+    let rec recursiveReverse list tempList =
+        match list with
+        | [] -> tempList
+        | h :: t -> recursiveReverse t (h :: tempList)
+    recursiveReverse list []
+
+/// Checking is string a palindrom
+let pal str =
+    let list = [for i in str -> i]
+    let rec innerRec list =
+        match list with 
+        | [] -> true
+        | h :: [] -> true
+        | h :: t -> if h = (reverseList t).Head then innerRec (reverseList t).Tail else false
```

Comment:
Все, понял, в чем проблема, пофиксил С:

2016-03-19 22:38 GMT+03:00 Yurii Litvinov <notifications@github.com>:

> In 4semester/26022016/26022016/Functions.fs
> <https://github.com/likeanowl/homeworks/pull/1#discussion_r56753990>:
>
> > +/// Reverses list (helper function)
> > +let reverseList list =
> > +    let rec recursiveReverse list tempList =
> > +        match list with
> > +        | [] -> tempList
> > +        | h :: t -> recursiveReverse t (h :: tempList)
> > +    recursiveReverse list []
> > +
> > +/// Checking is string a palindrom
> > +let pal str =
> > +    let list = [for i in str -> i]
> > +    let rec innerRec list =
> > +        match list with
> > +        | [] -> true
> > +        | h :: [] -> true
> > +        | h :: t -> if h = (reverseList t).Head then innerRec (reverseList t).Tail else false
>
> если на каждом шагу рекурсии вычислять обращение списка, то никакого
> профита
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/likeanowl/homeworks/pull/1/files/a26faa42e0e007fed82ce4c3c514504e51568a23#r56753990>
>



-- 

best wishes.

Svitkov Sergey.